[DOCS] Adds the release notes for 8.13.0 (#179216)
elastic/kibana: docs/CHANGELOG.asciidoc

Management::
* Adds a method of excluding data tiers when getting a field list ({kibana-pull}167946[#167946]).
Reporting::
* Adds support for 'auto' value for CSV scrolling duration ({kibana-pull}175005[#175005]).
* Adds CSV reporting with {esql} in Discover ({kibana-pull}174511[#174511]).
Observability::
* Adds actions column ({kibana-pull}175872[#175872]).
* Adds resource column with tooltip ({kibana-pull}175287[#175287]).
* Adds support for the Timeslice Metric visualization on the SLO detail page ({kibana-pull}175281[#175281]).
* Refactor alert table registration and change default columns ({kibana-pull}175119[#175119]).
* Adds customization for virtual columns and add the 1st virtual column ({kibana-pull}173732[#173732]).
* Adds a new option, Visualize this query, to the generated {esql} quires in the Elastic Assistant ({kibana-pull}174677[#174677]).

For more information about the features introduced in 8.13.0, refer to <<whats-new,What's new in 8.13>>.

[[enhancements-and-bug-fixes-v8.13.0]]
=== Enhancements and bug fixes

For detailed information about the 8.13.0 release, review the enhancements and bug fixes.

[float]
[[enhancement-v8.13.0]]

> === Enhancements

Reviewer: @TattdCodeMonkey @serenachou should we add a release note regarding embeddable console?

I think this is the initial PR: https://github.com/elastic/kibana/pull/174643

Author: Ah if it's Serverless releated I'm going to exclude that for now while it's gated.

* Adds table search to services, transactions and errors ({kibana-pull}174490[#174490]).
* Adds memoization to hooks consumed on service inventory page ({kibana-pull}173973[#173973]).
* Adds stack traces Threads embeddable ({kibana-pull}173905[#173905]).
* Enhances the diff topN functions ({kibana-pull}173397[#173397]).
* Updates Indices API to support sourcemap parameters ({kibana-pull}177847[#177847]).
* Fixes sorting instances table results on server-side ({kibana-pull}174164[#174164]).
Cases::
* Required custom fields now support default values, which will be used to automatically populate the custom fields if they are not defined when creating and updating cases ({kibana-pull}175961[#175961]).
* Persists all filter options of the cases table, including custom fields, in the URL. The filtering is also persisted when navigating back and forth between pages ({kibana-pull}175237[#175237]).
* Enables the alerts table for cases in the Stack Management ({kibana-pull}172217[#172217]).
Dashboard::
* Adds step setting for range slider control ({kibana-pull}174717[#174717]).
* Adds a deprecation badge in Dashboard on legacy control panels ({kibana-pull}174302[#174302]).

> * Adds external link icon ({kibana-pull}174407[#174407]).

Reviewer 2: Maybe we could give more context? "Adds external link icon to external URL links in the Links panel"

* Re-adds filtering settings in the Control settings UI ({kibana-pull}172857[#172857]).

> * Adds numeric options list ({kibana-pull}172106[#172106]).

Reviewer 2: A bit more context here might help. Maybe: "Adds number field support to the Options List control"

Discover::
* Hides the Empty fields section if there are no fields in it ({kibana-pull}172956[#172956]).

> * Adds tabs to Discover and Log Explorer ({kibana-pull}171467[#171467]).

Reviewer 3: This is just relevant for serverless, invisible elsewhere, should this be mentioned here? if yes, then I guess serverless should be mentioned?

Author: Ah no, I should take it out. Thanks for catching that :)

> * Adds header row height configuration ({kibana-pull}175501[#175501]).

Reviewer 3: ```suggestion
* Adds data table header row height configuration ({kibana-pull}175501[#175501]).
```
makes it a bit clearer

> * Adds new fields ingested in the background with valid mappings ({kibana-pull}172329[#172329]).

Reviewer 3: ```suggestion
* Adds new fields ingested in the background to the field list with valid mappings ({kibana-pull}172329[#172329]).
```
Maybe then it's clearer?

* AIOps: Adds link to log rate analysis from anomaly table ({kibana-pull}175289[#175289]).
* AIOps: Improves pattern analysis refresh behavior ({kibana-pull}174516[#174516]).
* Adds option for using table layout in the Change Point Detection embeddable ({kibana-pull}174348[#174348]).
* Adds high count option to the anomaly detection categorization wizard ({kibana-pull}174252[#174252]).
* Improves Data drift time range selection & shows hints for analysis process ({kibana-pull}174049[#174049]).
* Adds link to anomaly detection job creation from the alerting rule form ({kibana-pull}174016[#174016]).
* Adds warning for legacy method for installing pre-configured APM transaction job ({kibana-pull}173375[#173375]).
* Enhances toast notifications to improve error reporting ({kibana-pull}173362[#173362]).
* Adds document count chart for file upload ({kibana-pull}173210[#173210]).
* Trained models: Adds workflow for creating an ingest pipeline for a trained model ({kibana-pull}170902[#170902]).
* Updates alerts-as-data payload for Anomaly detection health and Transform health rules ({kibana-pull}176307[#176307]).
Management::

> * Change the column components in the index templates table ({kibana-pull}175823[#175823]).

Reviewer: ```suggestion
* Changes the column "Components" in the index templates table to display number of component templates ({kibana-pull}175823[#175823]).
```

* Input fields to change snapshot name and repository are now disabled when editing managed SLM policies in Snapshot & Restore ({kibana-pull}172291[#172291]).
Maps::
* Fixes Request URL Too Long (414) with heatmap layer when data view has larger number of date fields ({kibana-pull}177900[#177900]).
* Fixes the maps application breaking if you open a map with layers or sources that do not exist (#176419).
Observability::
* SLOs: Does not display group by cardinality when group by is not selected ({kibana-pull}178133[#178133]).
* Fixes bug in inventory rule for Inbound and Outbound traffic threshold (both preview and executor) ({kibana-pull}177997[#177997]).
* Fixes more lenient parsing of suggestion scores ({kibana-pull}177898[#177898]).
* Fixes refreshing the conversations list on conversation update ({kibana-pull}177897[#177897]).
* Fixes SLO details path is broken when `instanceId` contains a forward slash ({kibana-pull}177843[#177843]).
* Prevents users from picking date fields for the group-by selector ({kibana-pull}177830[#177830]).
* Fixes undefined issue cased due row check missing ({kibana-pull}177293[#177293]).

> * Reverts changes to `useLicense` hook ({kibana-pull}174372[#174372]).

Reviewer 3: this just caught my attention because of the title, not our team, but a revert title points to something that maybe didn't make it into production, needs to be double checked, but this could be the case, since this was reverted: https://github.com/elastic/kibana/pull/174031

* Adds threshold to the custom threshold alert document ({kibana-pull}176043[#176043]).
* Adds the ability to post Block Kit messages to the Slack Web API action ({kibana-pull}174303[#174303]).
* Adds criticality fields and risk score fields to alert schema ({kibana-pull}174626[#174626]).
* Adds fields table to rule details page alert flyout ({kibana-pull}172830[#172830]).
APM::
* Show Universal Profiling on Transaction view ({kibana-pull}176302[#176302]).
* Adds a specific metrics dashboard for opentelemetry-node services ({kibana-pull}174700[#174700]).
Elastic Security::
For the Elastic Security 8.13.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].
Fleet::
* Adds reference to `ecs@mappings` for each index template ({kibana-pull}174855[#174855]).
* Adds support for the `subobjects` setting on the object type mapping ({kibana-pull}171826[#171826]).
Infrastructure::
* Adds a new Services component to host details UI ({kibana-pull}176539[#176539]).
Integrations::
* Adds tiles for Notion and Redis connectors ({kibana-pull}177306[#177306]).
Lens & Visualizations::

> * Adds the creating of {esql} charts from the dashboard ({kibana-pull}171973[#171973]).

Reviewer 4: Let's move this to the dashboard section

> * Adds enhanced {esql} query editing experience with client side validation ({kibana-pull}170071[#170071]).

Reviewer 4: Let's create a new ES|QL section and add this because this is a feature which corresponds everywhere the ES|QL editor exists and not only on visualizations

Author: That makes a lot of sense, thank you!

* Adds stack traces Threads embeddable ({kibana-pull}173905[#173905]).
* Enhances the diff topN functions ({kibana-pull}173397[#173397]).
* Updates Indices API to support sourcemap parameters ({kibana-pull}177847[#177847]).
* Fixes sorting instances table results on server-side ({kibana-pull}174164[#174164]).
Cases::
* Adds support default values for required custom fields ({kibana-pull}175961[#175961]).
* Collects custom fields telemetry metrics for cases plugin in {kib} ({kibana-pull}175425[#175425]).
* Persists all filter options of the cases table in the URL ({kibana-pull}175237[#175237]).
Dashboard::
* Adds step setting for range slider control ({kibana-pull}174717[#174717]).
* Adds a deprecation badge in Dashboard on legacy control panels ({kibana-pull}174302[#174302]).
* Adds external link icon ({kibana-pull}174407[#174407]).
* Re-adds filtering settings in the Control settings UI ({kibana-pull}172857[#172857]).
* Adds numeric options list ({kibana-pull}172106[#172106]).
Discover::
* Hides the Empty fields section if there are no fields in it ({kibana-pull}172956[#172956]).
* Adds tabs to Discover and Log Explorer ({kibana-pull}171467[#171467]).

> * Adds multiple improvements to saved query management ({kibana-pull}170599[#170599]).

Reviewer 4: This is not discover specific. It happens anywhere the Unified search component exists. So I would move it to a new Unified search section

Elastic Search::
* Adds `ignore_empty_value` to generated set processor ({kibana-pull}175172[#175172]).
Fleet::
* Adds `skipRateLimitCheck` flag to the Upgrade API and Bulk_upgrade API ({kibana-pull}176923[#176923]).
* Adds making datastream rollover lazy ({kibana-pull}176565[#176565]).
* Stops creating  the `{type}-{datastet}@custom` component template during package installation ({kibana-pull}175469[#175469]).
* Adds the `xpack.fleet.isAirGapped` flag ({kibana-pull}174214[#174214]).
* Adds a warning when download upgrade is failing ({kibana-pull}173844[#173844]).
* Adds a message explaining why an agent is not upgradeable ({kibana-pull}173253[#173253]).
* Makes logs-* and metrics-* data views available across all spaces ({kibana-pull}172991[#172991]).
* Adds flag for pre-release to templates/inputs endpoint ({kibana-pull}174471[#174471]).
* Adds concurrency control to Fleet data stream API handler ({kibana-pull}174087[#174087]).
* Adds a handlebar helper to percent encode a given string ({kibana-pull}173119[#173119]).
Integrations::
* Enables minute frequency for incremental syncs ({kibana-pull}176603[#176603]).
Lens & Visualizations::

> * Adds clickable badges on compact view for {esql} queries ({kibana-pull}176568[#176568]).

Reviewer 4: This should be moved to a new ES|QL section as it is about the eS|QL editor and not specific to Visualizations

> * Adds link on the documentation popover to navigate to our external docs ({kibana-pull}176377[#176377]).

Reviewer 4: Same here.

* Adds support for canceling {esql} queries in *Lens* ({kibana-pull}176277[#176277]).

> * Adds quick fixes feature to {esql} query validation errors ({kibana-pull}175553[#175553]).

Reviewer 4: Same here

> * Adds support for command settings ({kibana-pull}175114[#175114]).

Reviewer 4: Same here

* Datatable improvements in *Lens* ({kibana-pull}174994[#174994]).

> * Adds 'greater than or equals to' and 'less than' options to filter options for date ranges and numbers ({kibana-pull}174283[#174283]).

Reviewer 4: This is not visualizations specific. It happens anywhere the Unified search component exists. So I would move it to a new Unified search section

Author: Ok brilliant, will do.

> * Adds line breaks redesign ({kibana-pull}173596[#173596]).

Reviewer 4: This too, let's also change it to

```
Allows line breaks on the ES|QL editor
```

* Adds IP Prefix Aggregation-based Visualization ({kibana-pull}173474[#173474]).

> * Adds a one minute option to the date picker ({kibana-pull}172944[#172944]).

Reviewer 4: This is not visualizations specific. It happens anywhere the Unified search component exists. So I would move it to a new Unified search section

Machine Learning::
* Adds the single metric viewer embeddable for dashboards ({kibana-pull}175857[#175857]).
* Adds support for {esql} in Data visualizer ({kibana-pull}174188[#174188]).
Management::
* Adds a method of excluding data tiers when getting a field list ({kibana-pull}167946[#167946]).
Reporting::
* Adds support for 'auto' value for CSV scrolling duration ({kibana-pull}175005[#175005]).
* Adds CSV reporting with {esql} in Discover ({kibana-pull}174511[#174511]).
Observability::
* Adds actions column ({kibana-pull}175872[#175872]).
* Adds resource column with tooltip ({kibana-pull}175287[#175287]).
* Adds support for the Timeslice Metric visualization on the SLO detail page ({kibana-pull}175281[#175281]).
* Refactor alert table registration and change default columns ({kibana-pull}175119[#175119]).
* Adds customization for virtual columns and add the 1st virtual column ({kibana-pull}173732[#173732]).
* Adds a new option, Visualize this query, to the generated {esql} quires in the Elastic Assistant ({kibana-pull}174677[#174677]).
Security::
* Adds context information in Risk Summary ({kibana-pull}174574[#174574]).

> * Adds the Asset Criticality selector for new entity flyouts ({kibana-pull}174438[#174438]).

Reviewer 5: Would these belong as part of the `{security-guide}/release-notes.html[_Elastic Security Solution Release Notes_]`?

* Adds link for AI Assistant in Observability left hand navigation ({kibana-pull}176144[#176144]).
* Handle token limit error message improvement ({kibana-pull}175871[#175871]).
* Corrects common {esql} mistakes ({kibana-pull}175520[#175520]).
* Adds resource column with tooltip ({kibana-pull}175287[#175287]).
* Adds alert fields table to Observability flyout ({kibana-pull}174685[#174685]).
Platform::
* Adds an option to disable APM user redaction ({kibana-pull}176566[#176566]).
Reporting::
* Updated CSV export to insert error messages into the contents if the export results in an empty file due to an error ({kibana-pull}175852[#175852]).
* Adds setting to use PIT or Scroll API ({kibana-pull}174980[#174980]).
Security::
* Default value of `server.securityResponseHeaders.referrerPolicy` changed to `strict-origin-when-cross-origin` ({kibana-pull}177559[#177559]).
* Adds the asset criticality UI ({kibana-pull}173512[#173512]).
* Show Asset Criticality in Risk Input flyout ({kibana-pull}174638[#174638]).
* Restrict management page based on user privileges ({kibana-pull}173178[#173178]).
* Merge risk input and asset integration expandable panels ({kibana-pull}173028[#173028]).

> * Require full risk engine privileges to enable, disable and init the risk engine ({kibana-pull}172932[#172932]).

Reviewer 5: These are part of the Security Solution, as opposed to stack security features.

* Fixes the maps application breaking if you open a map with layers or sources that do not exist (#176419).
Observability::
* SLOs: Does not display group by cardinality when group by is not selected ({kibana-pull}178133[#178133]).
* Fixes bug in inventory rule for Inbound and Outbound traffic threshold (both preview and executor) ({kibana-pull}177997[#177997]).
* Fixes more lenient parsing of suggestion scores ({kibana-pull}177898[#177898]).
* Fixes refreshing the conversations list on conversation update ({kibana-pull}177897[#177897]).
* Fixes SLO details path is broken when `instanceId` contains a forward slash ({kibana-pull}177843[#177843]).
* Prevents users from picking date fields for the group-by selector ({kibana-pull}177830[#177830]).
* Fixes undefined issue cased due row check missing ({kibana-pull}177293[#177293]).
* Reverts changes to `useLicense` hook ({kibana-pull}174372[#174372]).
* Fixes making IDs unique to capture multiple invocations of the same query ({kibana-pull}173433[#173433]).
Querying & Filtering::
* Fixes autocomplete value suggestions for KQL when the corresponding index has no tier preference set ({kibana-pull}176355[#176355]).
Security::

> * Prevents multiple duplicated calls to index_status API on alert details page ({kibana-pull}176573[#176573]).

Reviewer 5: These are a part of the security solution

Author: Ah ok sorry about that, I'll remove them and just check my release notes tool config.

Reviewer 5: No need to apologize, it's confusing to say the least 😅

* Adds resource column with tooltip ({kibana-pull}175287[#175287]).
* Adds support for the Timeslice Metric visualization on the SLO detail page ({kibana-pull}175281[#175281]).
* Refactor alert table registration and change default columns ({kibana-pull}175119[#175119]).
* Adds customization for virtual columns and add the 1st virtual column ({kibana-pull}173732[#173732]).
* Adds a new option, Visualize this query, to the generated {esql} quires in the Elastic Assistant ({kibana-pull}174677[#174677]).
Security::
* Adds context information in Risk Summary ({kibana-pull}174574[#174574]).
* Adds the Asset Criticality selector for new entity flyouts ({kibana-pull}174438[#174438]).

For more information about the features introduced in 8.13.0, refer to <<whats-new,What's new in 8.13>>.

[[enhancements-and-bug-fixes-v8.13.0]]
=== Enhancements and bug fixes

For detailed information about the 8.13.0 release, review the enhancements and bug fixes.

[float]
[[enhancement-v8.13.0]]
=== Enhancements
Alerting::
* Improve default AlertsTable columns configuration ({kibana-pull}176137[#176137]).
* Evenly distribute bulk-enabled alerting rules ({kibana-pull}172742[#172742]).

> * Enables the alerts table for cases in the Stack Management ({kibana-pull}172217[#172217]).

Reviewer 6: I think it is better to move this to the `Cases` section.

* Implement a mechanism to copy source data into the alerts-as-data documents for ES Query rules ({kibana-pull}171129[#171129]).
APM::
* Adds detectors for anomaly rules creation ({kibana-pull}171901[#171901]).
* Allows Universal Profiling agent to send error frames ({kibana-pull}176537[#176537]).
* Adds Azure settings ({kibana-pull}176386[#176386]).
* Adds table search to services, transactions and errors ({kibana-pull}174490[#174490]).
* Adds memoization to hooks consumed on service inventory page ({kibana-pull}173973[#173973]).
* Adds stack traces Threads embeddable ({kibana-pull}173905[#173905]).
* Enhances the diff topN functions ({kibana-pull}173397[#173397]).
* Updates Indices API to support sourcemap parameters ({kibana-pull}177847[#177847]).
* Fixes sorting instances table results on server-side ({kibana-pull}174164[#174164]).
Cases::

> * Adds support default values for required custom fields ({kibana-pull}175961[#175961]).

Reviewer 6: The release note is different from the PR. The feature changed during the development. It should be "Required custom fields now support default values. These default values will be used to populate automatically the custom fields if they are not defined when creating and updating cases."

Author: Do you mind if I change is slightly @cnasikas  to "Required custom fields now support default values, which will be used to automatically populate the custom fields if they are not defined when creating and updating cases ({kibana-pull}175961[#175961])."? 
Just to keep it to one sentence to be uniform with the other release notes. No worries if not just checking.

Reviewer 6: Ofc! You know better than me how we should phrase it properly 🙂. Whatever you think is best.

> * Collects custom fields telemetry metrics for cases plugin in {kib} ({kibana-pull}175425[#175425]).

Reviewer 6: We should remove this as it is not exposed to the users.

Author: Thanks for catching this

> * Persists all filter options of the cases table in the URL ({kibana-pull}175237[#175237]).

Reviewer 6: Also, this changed. It should be "Persist all filter options of the cases table, including custom fields, in the URL. The filtering is also persisted when navigating back and forth between pages."

Unified Search::
* Adds 'greater than or equals to' and 'less than' options to filter options for date ranges and numbers ({kibana-pull}174283[#174283]).
* Adds a one minute option to the date picker ({kibana-pull}172944[#172944]).
* Adds multiple improvements to saved query management ({kibana-pull}170599[#170599]).

[float]
[[fixes-v8.13.0]]
=== Bug Fixes
Alerting::
* Fixes Elasticsearch query rule with KQL evaluation matched document count ({kibana-pull}176620[#176620]).
* Fixes alerts not being visible when number of alerts are more than max alert limit ({kibana-pull}178019[#178019]).
* Fixes `ruleClient.getAlertState` error when a task is no longer available ({kibana-pull}177077[#177077]).
* Fixes AlertsTable sorting with inactive columns and default sort direction ({kibana-pull}176172[#176172]).
APM::
* Fixes an infinite loop caused by matching child and parent IDs ({kibana-pull}177914[#177914]).
* Fixes inconsistencies on Service Overview page ({kibana-pull}176293[#176293]).
* Fixes occurrences cut off value to be fully visible ({kibana-pull}175307[#175307]).
Cases::

> * Update UI to use custom field internal API ({kibana-pull}176574[#176574]).

Reviewer 6: I noticed that we forgot to add a release note section to this PR https://github.com/elastic/kibana/pull/176574. I updated it. It should be "Fix a bug where If there are required custom fields on a case whose values are empty it is not possible to update any of those fields.".
